# **Hands-on course, The Faculty of Informatics, Masaryk University Brno, November 2025**

This readme was created by Jiří Vorel (vorel@cesnet.cz).

Jiří Vorel lectured the course.

The information given in this course is current as of 4 November 2025.

<p align="center"><img src="https://github.com/CESNET/metacentrum-hands-on/blob/main/figures/002_fi_muni.png"></p>

# Table of contents
* [Introduction](#introduction)
 	* [Aims](#aims)
	* [Prerequisites](#prerequisites)
	* [Dedicated resources](#dedicated-resources)
	* [Data and tools](#data-and-tools)

# Introduction

## Aims
This tutorial takes the form of a brief hands-on course showing how to process and analyse sequencing data using the [MetaCentrum NGI](https://www.metacentrum.cz/en/index.html) (National Grid Infrastructure). Participants will be introduced to the basics of using MetaCentrum, including how to [log in to the front-end server](https://docs.metacentrum.cz/en/docs/access/log-in), [manipulate data](https://docs.metacentrum.cz/en/docs/data/large-data), [start an interactive or batch job](https://docs.metacentrum.cz/en/docs/computing/run-basic-job) and [install applications](https://docs.metacentrum.cz/en/docs/software/install-software) locally.

In the practical section of the course, we will analyse private (unpublished) [Oxford Nanopore Technologies](https://nanoporetech.com/) (ONT) data from a pilot project investigating the distribution of methylation sites in the genome of [_Haemonchus contortus_](https://en.wikipedia.org/wiki/Haemonchus_contortus), a gastrointestinal parasite of small ruminants. Unfortunately, processing raw ONT reads, basecalling and working with modified bases often require considerable hardware resources and time-consuming tuning to find the optimal parameters. To speed up this course, therefore, we will use a reduced input dataset and mostly default parameters during data processing.

<p align="center"><img src="https://upload.wikimedia.org/wikipedia/commons/5/55/Haemonchus_contortus_%28YPM_IZ_101907%29.jpeg"></p>

> [!IMPORTANT]  
> **The aim of this course is not to create a perfect ONT data processing pipeline.**
> 
> Due to the time limitations of this course and its primary focus on using grid infrastructure as effectively as possible, we will carry out a few basic steps on a small sample of ONT genomic reads. The aim is to identify modified (methylated) bases.

> [!NOTE]
> Throughout this tutorial, important points are supplemented with links to more detailed descriptions and explanations in the documentation.

## Prerequisites

To get the most out of this course, each participant should
  -  **be a registered MetaCentrum user** (due to the account approval and creation process, applications must be submitted no later than one day before the start of the course).
  -  **know the login information** (username and password, created during the application process).
  -  **have a laptop with a working internet connection.**
  -  **be able to log in to the remote server via the SSH protocol.**
  -  **know how to use the command line interface (CLI)**, recommended.
  
> [!NOTE]
> This course is designed for participants without strong command line knowledge. However, some basic knowledge is recommended. All commands and shell scripts used during the course can be found below and copied directly.

> [!NOTE]
> No data or software tools need to be downloaded or installed by participants before the course. Data will be downloaded during the course, and all software tools (which are freely available for non-commercial use) are already accessible to MetaCentrum users. If not, we will install them.

## Dedicated resources

As is typical for grid computing, submitted jobs are sorted into specific [queues](https://docs.metacentrum.cz/en/docs/computing/concepts#queues), mainly based on the amount of requested resources. The delay between job submission and the start of the calculation is determined by the combination of the required resources and the current infrastructure load. Very demanding jobs may have to wait several days in the queue before all the required resources are free. To avoid this, we will use a special queue reserved exclusively for this course, `MetaSeminar`. This queue uses the [zenon60.cerit-sc.cz](https://metavo.metacentrum.cz/pbsmon2/machine/zenon60.cerit-sc.cz) compute node with 32 AMD EPYC 7351 CPU cores, 512 GB RAM and 2 TB SSD + 2 TB HDD scratch discs. 

> [!IMPORTANT]  
> Each job submitted during this course must target this dedicated queue. As you will see later, interactive jobs will include the parameter `-q MetaSeminar`, while batch jobs will include the line with `#PBS -q MetaSeminar`. The job scheduler [PBSPro](https://docs.metacentrum.cz/en/docs/computing/concepts#pbs-server) will send jobs to this specified queue in both cases.

## Data and tools

The following data and software tools will be used during the course:

 - Oxford Nanopore reads (will be provided during the course).
 - [POD5](https://github.com/nanoporetech/pod5-file-format) Python package for inspecting and manipulating POD5 files.
 - [Dorado](https://github.com/nanoporetech/dorado) for basecalling.
 - [Cramino](https://github.com/wdecoster/cramino) for quick quality assessment.
 - [Samtools](https://github.com/samtools/samtools) for handling BAM files.
 - [Modkit](https://github.com/nanoporetech/modkit) for working with modified bases from Oxford Nanopore.


